tests/tr1/tests/valarray/test.cpp

```diff
@@ -340,5 +340,24 @@ void test_main() { // test basic workings of valarray definitions
     CHECK_INT(v0.max(), 4);
     CHECK_INT(v0.min(), 2);
 
+    // test range-based for
+    v0 = Mycont();
+    for (char c : v0) {
+        (void) c;
+        CHECK(false);
+    }
+
+    v0 = Mycont("\1\1\1", 3);

```

Comment:
I was about to ask where you got this horrible style from, and then I saw this is a pre-existing tr1 test. (No change requested, just despair at the horrors that appear in some of our ancient tests.)

Reply:
Agreed - we don't generally modify the legacy `tr1` test suite, but this seems like a reasonable place to add coverage and there isn't a better place in `tests/std` without adding a new test there. Also no change requested.